src/github.com/matrix-org/go-neb/services/guggy/guggy.go

```diff
@@ -0,0 +1,135 @@
+package services
+
+import (
+	"encoding/json"
+	"bytes"
+	"math"
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/plugin"
+	"github.com/matrix-org/go-neb/types"
+	"net/http"
+	"strconv"
+	"strings"
+)
+type guggyQuery struct {
+	// "mp4" of "gif"
+	Format string `json:"format"`
+	// Query sentence
+	Sentence  string `json:"sentence"`
+}
+
+type guggyGifResult struct {
+	ReqID string `json:"reqId"`
+	GIF string `json:"gif"`
+	Width float64 `json:"width"`
+	Height float64 `json:"height"`
+}
+
+type guggyService struct {
+	id            string
+	serviceUserID string
+	APIKey        string
```

Comment:
No, it has to be public in order for `json.Marshal` to see it. Over the wire we should be consistently using `keys_with_underscores`.